akka-docs-dev/rst/java/code/docs/MigrationsJava.java

```diff
@@ -0,0 +1,123 @@
+package docs;
+
+import akka.japi.Pair;
+import akka.stream.*;
+import akka.stream.javadsl.*;
+import scala.Option;
+import scala.concurrent.Promise;
+import scala.runtime.BoxedUnit;
+
+public class MigrationsJava {
+
+    // This is compile-only code, no need for actually running anything.
+    public static ActorMaterializer mat = null;
+
+    public static void main(String[] args) {
+
+        Outlet<Integer> outlet = null;
+
+        Outlet<Integer> outlet1 = null;
+        Outlet<Integer> outlet2 = null;
+
+        Inlet<Integer> inlet = null;
+
+        Inlet<Integer> inlet1 = null;
+        Inlet<Integer> inlet2 = null;
+
+        Flow<Integer, Integer, BoxedUnit> flow = Flow.of(Integer.class);
+        Flow<Integer, Integer, BoxedUnit> flow1 = Flow.of(Integer.class);
+        Flow<Integer, Integer, BoxedUnit> flow2 = Flow.of(Integer.class);
+
+        Promise<Option<Integer>> promise = null;
+
+
+        {
+            Graph<SourceShape<Integer>, BoxedUnit> graphSource = null;
+            Graph<SinkShape<Integer>, BoxedUnit> graphSink = null;
+            Graph<FlowShape<Integer, Integer>, BoxedUnit> graphFlow = null;
+
+            //#flow-wrap
+            Source<Integer, BoxedUnit> source = Source.fromGraph(graphSource);
+            Sink<Integer, BoxedUnit> sink = Sink.fromGraph(graphSink);
+            Flow<Integer, Integer, BoxedUnit> aflow = Flow.fromGraph(graphFlow);
+            Flow.fromSinkAndSource(Sink.<Integer>head(), Source.single(0));
+            Flow.fromSinkAndSourceMat(Sink.<Integer>head(), Source.single(0), Keep.left());
+            //#flow-wrap
+
+            Graph<BidiShape<Integer, Integer, Integer, Integer>, BoxedUnit> bidiGraph = null;
+
+            //#bidi-wrap
+            BidiFlow<Integer, Integer, Integer, Integer, BoxedUnit> bidiFlow =
+                BidiFlow.fromGraph(bidiGraph);
+            BidiFlow.fromFlows(flow1, flow2);
+            BidiFlow.fromFlowsMat(flow1, flow2, Keep.both());
+            //#bidi-wrap
+
+        }
+
+        {
+            //#graph-create
+            FlowGraph.create(builder -> {
+                //...
+                return ClosedShape.getInstance();
+            });
+
+            FlowGraph.create(builder -> {
+                //...
+                return new FlowShape<>(inlet, outlet);
```

Comment:
I was proposing adding one, similarily like `Flow.of()` exists already (and various other java apis, to avoid having to write the diamond (Pair etc)).